modules/mobile-agents/windows/jax-rs/src/main/java/org/wso2/carbon/mdm/mobileservices/windows/services/discovery/impl/DiscoveryServiceImpl.java

```diff
@@ -109,4 +100,37 @@ public Response discoverGet() {
         }
         return Response.ok().build();
     }
+
+    /**
+     * Get authentication policy from the tenant configuration, otherwise set default value as Federated.
+     *
+     * @return Authentication policy.
+     * @throws WindowsDeviceEnrolmentException
+     */
+    public String getAuthPolicy() throws WindowsDeviceEnrolmentException {
+        String authPolicy = null;
+        List<ConfigurationEntry> tenantConfigurations;
+        try {
+            if (WindowsAPIUtils.getTenantConfigurationData() != null) {
+                tenantConfigurations = WindowsAPIUtils.getTenantConfigurationData();
+
+                for (ConfigurationEntry configurationEntry : tenantConfigurations) {
+                    if (configurationEntry.getName().equals(PluginConstants.TenantConfigProperties.AUTH_POLICY)) {
```

Comment:
As a best practice, compare all variable values against known constants, which will let you get rid of redundant null checks to check if the value assigned to the corresponding variables is null.